src/drivers/protocol_splitter/protocol_splitter.cpp

```diff
@@ -218,23 +226,35 @@ int ReadBuffer::read(int fd)
 	return r;
 }
 
-void ReadBuffer::copy(void *dest, size_t pos, size_t n)
+size_t ReadBuffer::copy(void *dest, size_t pos, size_t n)
 {
-	ASSERT(pos < buf_size);
-	ASSERT(pos + n <= buf_size);
+	DEBUGASSERT(pos < buf_size);
+	DEBUGASSERT(pos + n <= buf_size);
 
-	if (dest) {
+	if (pos < buf_size && pos + n <= buf_size) {
```

Comment:
need to check dest pointer also:
`if (dest && pos < buf_size && pos + n <= buf_size) {`